packages/core/bootstrap/src/lib/external-adapter/validator.ts

```diff
@@ -19,25 +19,42 @@ import presetIncludes from './overrides/presetIncludes.json'
 import { Requester } from './requester'
 import { inputParameters } from './builder'
 
+const sharedInputConfig = excludableInternalAdapterRequestProperties.reduce<Record<string, false>>(
+  (config, name) => {
+    config[name] = false
+    return config
+  },
+  {},
+)
+
 export type OverrideType = 'overrides' | 'tokenOverrides' | 'includes'
+
+type InputType = {
+  id?: string
+  data?: any
+}
 export class Validator {
-  input: {
-    id?: string
-    data?: any
-  }
+  input: InputType
   inputConfigs: InputParameters
   options: Record<string, any[]>
   validated: any
   error: AdapterError | undefined
   errored: AdapterErrorResponse | undefined
   shouldLogError: boolean
 
-  constructor(input = {}, inputConfigs = {}, options = {}, shouldLogError = true) {
+  constructor(
+    input: InputType = { id: '1', data: {} },
+    inputConfigs = {},
+    options = {},
+    shouldLogError = true,
+  ) {
     this.input = { ...input }
-    this.inputConfigs = { ...inputConfigs }
+    if (!this.input.id) this.input.id = '1' //TODO Please remove these once "no any" strict typing is enabled
```

Comment:
when will we enable "no any" strict typing?  Also is there a Shortcut ticket for this so that we don't forget?

Reply:
Good point, I'll add a note in https://app.shortcut.com/chainlinklabs/story/11106/disable-allowing-of-any-for-new-adapters